src/Scenes/Opening.cpp

```diff
@@ -163,13 +174,72 @@ void Opening::initTraffic()
 
 std::uint16_t Opening::poll(const sf::Event& e)
 {
+
     if (e.type == sf::Event::KeyPressed)
     {
+        updateMessage();
+
         switch (e.key.code)
         {
             default:
             break;
 
+            //
+            // Action. Perform an action. E.g. pick up an item.
+            //
+            case sf::Keyboard::A:
+            {
+                //
+                // Check if we are on an item.
+                // If yes, remove it from the map's _items and add it to the 
+                // player's inventory.
+                //
+                for(auto it = _items.begin(); it != _items.end(); it++)
+                {
+                    ItemPtr item = *it;
+
+                    if( item->getGlobalBounds().intersects(
+                                                _player->getGlobalBounds()) &&
+                        item->isObtainable() )
+                    {
+
+                        _player->addItem(item->getID());
+                        _missionText.setText("Picked up " + item->getName());
+                        _items.erase(it);
+
+                        break;
+                    }
+                }
+            }
+            break;
+
+            //
+            // Inventory. Display inventory.
+            //
+            case sf::Keyboard::I:
+            {
+                std::map<std::string, std::int32_t> inv =
```

Comment:
If you make the changes above that I recommended you will need to change this, else you'll still make a copy. If you implement the read-only/const version, you should update this to:

`const std::map<...>& inv = ...`

Or simply: `const auto& inv = ...`

Reply:
Question:

When I change this to use a reference, this seems to break the use of the [] operator on the map.
A workaround I guess is to use map's at() method. Is there a better way to do this?

```
/home/amb/git-clones/charmyt.ttvg/src/Scenes/Opening.cpp:242:56: error: passing ‘const std::map<std::basic_string<char>, int>’ as ‘this’ argument discards qualifiers [-fpermissive]
  242 |                     std::cout << key << ": " << inv[key] << std::endl;
      |                                                        ^

...

  490 |       operator[](const key_type& __k)
      |       ^~~~~~~~

```

Reply:
The `at()` method was specifically added (in C++11 I think?) to address this situation: to allow const access to the elements.